tests/unit/test_column_noise.py

```diff
@@ -643,9 +644,93 @@ def test_generate_phonetic_errors():
+)
+def test_generate_ocr_errors(dummy_dataset, column):
+    data = dummy_dataset[column]
+    # Update column name for noise function
+    data.name = "first_name"
+
+    config = get_configuration()
+    config.update(
+        {
+            DATASETS.census.name: {
+                Keys.COLUMN_NOISE: {
+                    column: {
+                        NOISE_TYPES.ocr.name: {
+                            Keys.CELL_PROBABILITY: 0.1,
+                            Keys.TOKEN_PROBABILITY: 1.0,
+                        },
+                    },
+                },
+            },
+        }
+    )
+    config = config[DATASETS.census.name][Keys.COLUMN_NOISE][column][NOISE_TYPES.ocr.name]
+    noised_data = NOISE_TYPES.ocr(data, config, RANDOMNESS0, "test_ocr")
+
+    # Validate we do not change any missing data
+    missing_mask = data == ""
+    assert (data[missing_mask] == noised_data[missing_mask]).all()
+
+    # Check expected noise level
+    token_prob = config[Keys.TOKEN_PROBABILITY]
+    cell_prob = config[Keys.CELL_PROBABILITY]
+
+    check_original = data[~missing_mask]
+    check_noised = noised_data[~missing_mask]
+    str_lengths = check_original.str.len() + check_original.str.len() - 1
+    p_token_not_noised = 1 - token_prob  # pd.Series
+    # Get probability no tokens are noised in a string
+    p_strings_not_noised = p_token_not_noised**str_lengths  # pd.Series
+    p_strings_noised = 1 - p_strings_not_noised  # pd.Series
+    expected_noise = cell_prob * p_strings_noised.mean()
+    actual_noise = (check_original != check_noised).mean()
+    # We have simplified the expected noise calculation. Note that not all tokens are eligible for OCR noising so we
+    # should never meet our upper bound of expected noise. Alternatively, we want to make sure the noise level is not
+    # unexpectedly small.
+    assert actual_noise < expected_noise
+    assert actual_noise > expected_noise / 10
+
+
+def test_ocr_replacement_values():
+    # Test that OCR noising replaces truth value with correct error values
+    # Load OCR errors dict
+    ocr_errors_dict = load_ocr_errors_dict()
+    # Make series of OCR error dict keys - is there an intelligent numberto pick besides 10?
+    data = pd.Series(list(ocr_errors_dict.keys()) * 10, name="employer_name")
+    config = get_configuration()
+    config.update(
+        {
+            DATASETS.census.name: {
+                Keys.COLUMN_NOISE: {
+                    "employer_name": {
+                        NOISE_TYPES.ocr.name: {
+                            Keys.CELL_PROBABILITY: 1.0,
+                            Keys.TOKEN_PROBABILITY: 1.0,
+                        },
+                    },
+                },
+            },
+        }
+    )
+    config = config[DATASETS.census.name][Keys.COLUMN_NOISE]["employer_name"][
+        NOISE_TYPES.ocr.name
+    ]
+    noised_data = NOISE_TYPES.ocr(data, config, RANDOMNESS0, "test_ocr_error_values")
+
+    for key in ocr_errors_dict.keys():
```

Comment:
I probably wouldn't do this b/c it's kinda hard to read, but this _could_ be reduced to something like:
```
assert all(
    [
        x[0] in x[1]
        for x in zip(
            noised_data.employer_name,
            data.employer_name.map(ocr_error_dict)
        )
    ]
)
```